### DISCUSSION TITLE
JSON parse error - Expecting value: line 1 column 1 (char 0)"}*

### DISCUSSION THREAD

**Discussion by ughstudios** [nlohmann/json#3297] [Q&A] [answered]
```
{"detail":"JSON parse error - Expecting value: line 1 column 1 (char 0)"}* STATE: PERFORMING => DONE handle 0x243352a61a8; line 2465 (connection #0)
```

Here's my json:
``
`nlohmann::json json_data;
	json_data["username"] = username;
	json_data["password"] = password;```

Here's how my call is happening:

```
nlohmann::json json_data;
	json_data["username"] = username;
	json_data["password"] = password;
	CURL* curl = curl_easy_init();
	CURLcode res;

	if (curl)
	{
		curl_easy_setopt(curl, CURLOPT_URL, GetRegistrationURL().c_str());
		curl_easy_setopt(curl, CURLOPT_VERBOSE, 1L);
		
		struct curl_slist* chunk = NULL;
		chunk = curl_slist_append(chunk, "Content-Type: application/json");

		curl_easy_setopt(curl, CURLOPT_HTTPHEADER, chunk);
		curl_easy_setopt(curl, CURLOPT_POST, 1L);

		/* Set the expected POST size. */
		curl_easy_setopt(curl, CURLOPT_POSTFIELDSIZE, sizeof(json_data));
		curl_easy_setopt(curl, CURLOPT_POSTFIELDS, json_data.dump().c_str());
		curl_easy_setopt(curl, CURLOPT_USERAGENT, "Jr-1");


		res = curl_easy_perform(curl);
		if (res == CURLE_OK)
		{
			//res = curl_easy_getinfo(curl, CURLINFO_CONTENT_TYPE, &ct);
		}
		curl_easy_cleanup(curl);
		curl_slist_free_all(chunk);
	}

	curl_global_cleanup();

```

**Comment by nlohmann** [comment]
There is no call to parse in your code.

**Comment by gregmarr** [comment, marked as answer]
This error message isn't coming from this library, probably from the service that you are calling.

    curl_easy_setopt(curl, CURLOPT_POSTFIELDS, json_data.dump().c_str());

`json::dump()` returns a `std::string` which is a temporary.  You're extracting a `char *` from that and passing it to `curl_easy_setopt`, and then the temporary is destroyed, invalidating the pointer.  Try this instead:

    auto json_str = json_data.dump();
    curl_easy_setopt(curl, CURLOPT_POSTFIELDS, json_str.c_str());

**Reply by nlohmann** [reply]
@ughstudios Do you need further assistance with this issue?
